### PR TITLE
Add apple-ascl license

### PR DESCRIPTION
**Pull request by merveenoyan**
@osanseviero @pcuenca

### CODE REVIEW

File: docs/hub/repositories-licenses.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ A full list of the available licenses is available here:
 | ISC                                                            | `isc`                                    |
 | LaTeX Project Public License v1.3c                             | `lppl-1.3c`                              |
 | Microsoft Public License                                       | `ms-pl`                                  |
+| Apple ML-MobileCLIP License                                    | `ml-mobileclip`                          |
```
Comment (julien-c) [inline docs/hub/repositories-licenses.md:65]: needs an info: link or license: file

Comment (osanseviero) [inline docs/hub/repositories-licenses.md:65]: This is not a mobileCLIP specific license, it's a more general one. Apple folks called it `apple-ascl` at https://huggingface.co/apple/MobileCLIP-S1

**Review by osanseviero — CHANGES_REQUESTED** [review]

File: docs/hub/repositories-licenses.md (hunk 1 above)
Comment (pcuenca) [reply docs/hub/repositories-licenses.md:65]: Indeed, it's the "[Apple Sample Code License](https://developer.apple.com/support/downloads/terms/apple-sample-code/Apple-Sample-Code-License.pdf)".

Comment (merveenoyan) [reply docs/hub/repositories-licenses.md:65]: I could swear I saw ML-MobileCLIP license (which is MIT license with a disclaimer for distribution at the bottom) in TiC-CLIP repos. I'll fix thank you for the heads-up

Comment (merveenoyan) [reply docs/hub/repositories-licenses.md]: Fixed and added file

File: docs/hub/repositories-licenses.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ A full list of the available licenses is available here:
 | ISC                                                            | `isc`                                    |
 | LaTeX Project Public License v1.3c                             | `lppl-1.3c`                              |
 | Microsoft Public License                                       | `ms-pl`                                  |
+| Apple Sample Code license                                      | `apple-ascl`                             | <!-- license: https://github.com/apple/ml-tic-clip/blob/main/LICENSE -->
```
Comment (pcuenca) [inline docs/hub/repositories-licenses.md:65]: It's unfortunate that it's a PDF, but maybe `info: https://developer.apple.com/support/downloads/terms/apple-sample-code/Apple-Sample-Code-License.pdf` would ensure more longevity? (Repo licenses have changed in the past in several cases)

Comment (julien-c) [reply docs/hub/repositories-licenses.md:65]: pdf is fine IMO

Comment (osanseviero) [inline docs/hub/repositories-licenses.md:65]: Let's use this as per the comments
```suggestion
| Apple Sample Code license                                      | `apple-ascl`                             | <!-- license: https://developer.apple.com/support/downloads/terms/apple-sample-code/Apple-Sample-Code-License.pdf -->
```

**Review by pcuenca — APPROVED** [review]

**Review by julien-c — APPROVED** [review]

**Comment by merveenoyan** [comment]
@osanseviero I need your approval still

**Review by osanseviero — APPROVED** [review]

**Pull request merged**